src/scroll-timeline-css.js

```diff
@@ -144,13 +144,34 @@ export function initCSSPolyfill() {
 
   initMutationObserver();
 
+  // Cache all Proxy Animations
+  let proxyAnimations = new WeakMap();
+
   // We are not wrapping capturing 'animationstart' by a 'load' event,
   // because we may lose some of the 'animationstart' events by the time 'load' is completed.
   window.addEventListener('animationstart', (evt) => {
     evt.target.getAnimations().filter(anim => anim.animationName === evt.animationName).forEach(anim => {
-      const result = createScrollTimeline(anim, anim.animationName, evt.target);
-      if (result.timeline && anim.timeline != result.timeline) {
-        const proxyAnimation = new ProxyAnimation(anim, result.timeline, result.animOptions);
+      // Create a per-element cache
+      if (!proxyAnimations.has(evt.target)) {
+        proxyAnimations.set(evt.target, new Map());
+      }
+      const elementProxyAnimations = proxyAnimations.get(evt.target);
+
+      // Store Proxy Animation in the cache
+      if (!elementProxyAnimations.has(anim.animationName)) {
```

Comment:
`animation-name` allows listing the same animation more than once, e.g.
```css
#target {
  animation-name: spin, spin;
}
```
This is expected to produce 2 animations with the latter having a higher composite order, but I believe this change will result in a single animation.

Reply:
We'll need to track the index as well as the name.

Reply:
I created an alternative version in https://github.com/flackr/scroll-timeline/pull/89 which also implements getAnimations support in this polyfill.